src/Microsoft.IdentityModel.JsonWebTokens/JwtTokenUtilities.cs

```diff
@@ -85,6 +85,33 @@ public static string CreateEncodedSignature(string input, SigningCredentials sig
             }
         }
 
+        /// <summary>
+        /// Decompress JWT token bytes.
+        /// </summary>
+        /// <param name="tokenBytes"></param>
+        /// <param name="algorithm"></param>
+        /// <exception cref="ArgumentNullException">if 'tokenBytes' is null.</exception>
+        /// <exception cref="ArgumentNullException">if 'algorithm' is null.</exception>
+        /// <exception cref="NotSupportedException">if the decompression algorithm is not supported.</exception>
```

Comment:
Missing the InvalidOperationException.
I think there is a better choice, the application will not really know what to do here.
Maybe: SecurityTokenDecompressionFailedException (this would be new).

Reply:
Good catch! :) We actually already have a SecurityTokenDecompressionFailedException. I've made the change to throw this exception instead of InvalidOperationException.